apps/system/utils/fscmd.c

```diff
@@ -1097,15 +1114,14 @@ static int tash_rm(int argc, char **args)
 		ret = unlink(fullpath);
 		if (ret != OK) {
 			FSCMD_OUTPUT(CMD_FAILED, args[0], "unlink");
-			fscmd_free(fullpath);
-			return ret;
+			ret = -errno;
+			goto errout;
 		}
 		FSCMD_OUTPUT("%s deleted\n", fullpath);
 	}
 
-	if (fullpath) {
-		fscmd_free(fullpath);
-	}
+errout:
```

Comment:
Is the if(fullpath) check not necessary anymore since its been deleted ?

Reply:
Old comment on the same line.
fscmd_free(path) checks if path is NULL internally.